KBaseSearchEngine.spec

```diff
@@ -98,7 +98,7 @@ module KBaseSearchEngine {
       Search for number of objects of each type matching constraints.
     */
     funcdef search_types(SearchTypesInput params) 
-        returns (SearchTypesOutput) authentication required;
+        returns (SearchTypesOutput) authentication optional;
```

Comment:
The access filter definition in the spec needs to be changed. The current definition is, 

```
/*
      Optional rules of access constraints.
        - with_private - include data found in workspaces not marked 
            as public, default value is true,
        - with_public - include data found in public workspaces,
            default value is false,
        - with_all_history - include all versions (last one and all
            old versions) of objects matching constrains, default
            value is false.
    */
    typedef structure {
        boolean with_private;
        boolean with_public;
        boolean with_all_history;
    } AccessFilter;
```

Change 

```
- with_private - include data found in workspaces not marked 
            as public, default value is true for authenticated users and false for unauthenticated users,
- with_public - include data found in public workspaces,
            default value is false for authenticated users and true for unauthenticated users,
```

Reply:
Done